### DISCUSSION TITLE
Focus on interactive element inside of `ComboboxItem` when navigating with keyboard using arrow keys

### DISCUSSION THREAD

**Discussion by straku** [ariakit/ariakit#3985] [Questions] [answered]
I am working on this type of component using combobox:

<img width="495" alt="Screenshot 2024-08-02 at 16 59 47" src="https://github.com/user-attachments/assets/65499860-ffea-42cf-8800-62d84c3c231d">

When I search for something using the input I am able move to each of the combobox items (each of them is set correctly as active). Is it possible to circumvent the default behavior of what gets set to active and instead of having each `ComboboxItem`active make it focus on the button within it instead? 

Example could be helfpul. From the state where the previous screenshot was taken one click of arrow down would get me to:

<img width="492" alt="Screenshot 2024-08-02 at 17 06 13" src="https://github.com/user-attachments/assets/e5ffb0d7-d98f-4664-bdde-03063b7960a4">

And the second click of arrow down would get me to:

<img width="497" alt="Screenshot 2024-08-02 at 17 06 47" src="https://github.com/user-attachments/assets/8497df16-207c-40bf-ba6d-f2bbdb3c500b">

(etc., etc., etc.)

Each button that I want to focus on is also an Ariakit button. I've tried moving the focus to the button manually, but when I do that I am no longer able to move to a different `ComboboxItem` with up/down keys (which is understandable). I could manually manage moving the focus out of the button on key down/up events, but I wonder whether there might be an easier way. Any ideas?

**Comment by benrodrs** [comment, marked as answer]
`ComboboxItem` (in this case, an `option` element) shouldn't contain interactive elements inside it, as it is already an interactive element.

I suggest converting the button into a span element and styling it with focus using a selector like this:

```css
.combobox-item[data-active-item] .span {
  ...
}
```
